src/utils.js

```diff
@@ -40,3 +40,7 @@ export const getColorContrast = (relativeLuminance1, relativeLuminance2) => {
   const contrast = (lighterColor + 0.05) / (darkerColor + 0.05)
   return contrast
 }
+
+export const deepCopyArrayOfObject =(array)=>{
```

Comment:
I'm curious as to what benefits this has over just using the initial state as it is?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

Reply:
We can use the initial state but I wanted to not repeat the initial state so I created a variable in which I stored the initial state but once I initialize the colors state with initial State  and if we update the state by adding new field or changing the color of existing field the **initialState** variable is also updated because of shallow copy to avoid the shallow copy I create the function **deepCopyArrayOfObject** in utils file